### PR TITLE
python3Packages.pymoo: init at 0.6.0

### PR DESCRIPTION
**Pull request by veprbl**
###### Description of changes

Provides a library of algorithms for multi-objective optimization

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pbsds** [comment]
```shell
 nix-build . -A python39Packages.pymoo
these 2 derivations will be built:
  /nix/store/gnsablq2bf54v1mdgg1ikiqi48vz5n54-source.drv
  /nix/store/65f0vzlff0x90yn6036r7qpl3rwkhq2f-python3.9-pymoo-0.5.0.drv
building '/nix/store/gnsablq2bf54v1mdgg1ikiqi48vz5n54-source.drv'...

trying https://github.com/anyoptimization/pymoo/archive/0.5.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 8186k    0 8186k    0     0  1820k      0 --:--:--  0:00:04 --:--:-- 2125k
unpacking source archive /build/0.5.0.tar.gz
error: hash mismatch in fixed-output derivation '/nix/store/gnsablq2bf54v1mdgg1ikiqi48vz5n54-source.drv':
         specified: sha256-uS/gRHI8JrXspkms2hlIQNtMiF6N/pMkVmtBzZTMJ2Q=
            got:    sha256-tklnoiWOE5J67Y9DTIbUgUZgLbhDPp6mMfZ1XKhWFPw=
```

**Comment by veprbl** [comment]
@GrahamcOfBorg build python3Packages.pymoo

File: pkgs/development/python-modules/pymoo/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    repo = "pymoo";
+    rev = version;
+    sha256 = "sha256-uS/gRHI8JrXspkms2hlIQNtMiF6N/pMkVmtBzZTMJ2Q=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace "cma==2.7" "cma"
+    sed -i pymoo/experimental/constr.py \
+      -e '1ifrom pymoo.core.replacement import ReplacementSurvival'
```
Comment (jonringer) [inline pkgs/development/python-modules/pymoo/default.nix:26]: not a blocker for this PR, but I feel like this should be upstreamed

File: pkgs/development/python-modules/pymoo/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+buildPythonPackage rec {
+  pname = "pymoo";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "anyoptimization";
+    repo = "pymoo";
+    rev = version;
+    sha256 = "sha256-uS/gRHI8JrXspkms2hlIQNtMiF6N/pMkVmtBzZTMJ2Q=";
```
Comment (jonringer) [inline pkgs/development/python-modules/pymoo/default.nix:19]: is there some unicode?

```
error: hash mismatch in fixed-output derivation '/nix/store/gnsablq2bf54v1mdgg1ikiqi48vz5n54-source.drv':
         specified: sha256-uS/gRHI8JrXspkms2hlIQNtMiF6N/pMkVmtBzZTMJ2Q=
            got:    sha256-tklnoiWOE5J67Y9DTIbUgUZgLbhDPp6mMfZ1XKhWFPw=

File: pkgs/development/python-modules/pymoo/default.nix (hunk 1 above)
Comment (veprbl) [reply pkgs/development/python-modules/pymoo/default.nix]: fixed in 0.6.0

**Comment by veprbl** [comment]
@GrahamcOfBorg build python310Packages.pymoo

**Pull request merged**